docs/Customization/Custom-Core.rst

```diff
@@ -0,0 +1,389 @@
+provide two examples of how to integrate a core.
+
+.. note:: 
+
+    RoCC is not supported by custom core currently. Please use Rocket or Boom as the RoCC base core if you need to use RoCC.
+
+Parameter Case Classes
+----------------------
+
+Chipyard will generate a core for every ``InstantiableTileParams`` object it discovered in the current config.
+This object is derived from``TileParams``, a trait containing the information needed to create a tile. All cores must have
+their own implementation of ``InstantiableTileParams``, as well as ``CoreParams`` which is passed as a field in ``TileParams``.
+
+``TileParams`` holds the parameters for the tile, which are the same for every generated core, while ``CoreParams`` 
+contains the parameters for individual cores. They must be implemented as case classes with fields that can be overridden by 
+other config fragments as the constructor parameters. See the appendix at the bottom of the page for a list of 
+variable to be implemented. You can also add custom fields to them, but standard fields should always be preferred. 
+
+``InstantiableTileParams[TileType]`` holds the constructor of ``TileType`` on top of the fields of ``TileParams``.
+All custom cores will also need to implement ``instantiate()`` in their tile parameter class to return a new instance
+of the tile class ``TileType``. 
+
+``TileParams``, ``InstantiableTileParams[TileType]`` and ``CoreParams`` contains the following fields (you may ignore 
+any fields marked "Rocket specific" and use their default values, although it is recommended to use them if you 
+need a custom field with similar purposes):
+
+.. code-block:: scala
+
+    trait TileParams {
+      val core: CoreParams                  // Core parameters (see below)
+      val icache: Option[ICacheParams]      // Rocket specific: I1 cache option
+      val dcache: Option[DCacheParams]      // Rocket specific: D1 cache option
+      val btb: Option[BTBParams]            // Rocket specific: BTB / branch predictor option
+      val hartId: Int                       // Hart ID: Must be unique within a design config
+      val beuAddr: Option[BigInt]           // Rocket specific: Bus Error Unit for Rocket Core
+      val blockerCtrlAddr: Option[BigInt]   // Rocket specific: Bus Blocker for Rocket Core
+      val name: Option[String]              // Name of the core
+    }
+
+    abstract class InstantiableTileParams[TileType <: BaseTile] extends TileParams {
+      def instantiate(crossing: TileCrossingParamsLike, lookup: LookupByHartIdImpl)
+                    (implicit p: Parameters): TileType
+    }
+
+    trait CoreParams {
+      val bootFreqHz: BigInt              // Frequency
+      val useVM: Boolean                  // Support virtual memory
+      val useUser: Boolean                // Support user mode
+      val useSupervisor: Boolean          // Support supervisor mode
```

Comment:
Ideally this would be an `include` statement so the code doesn't turn stale

Reply:
Done. Used an example file to hold them.